client/src/app/components/signup-form/signup-form.component.html

```diff
@@ -22,14 +22,17 @@ <h1 class="signup-title">Sign Up</h1>
         </mat-form-field>
 
         <mat-form-field class="mt-input-field half" appearance="outline" color="accent">
-            <input id="signup-password-2" matInput placeholder="Confirm Password" type="password" [formControl]="pwd2" />
+            <input id="signup-password-2" matInput placeholder="Confirm Password" type="password"
+                [formControl]="pwd2" />
         </mat-form-field>
 
-        <button id="signup-submit-btn" class="mt-input-field" mat-flat-button type="submit" [class.btn-opacity-disabled]="signUpForm.invalid"
-            [disabled]="!signUpForm.valid">
+        <button id="signup-submit-btn" class="mt-input-field" mat-flat-button type="submit"
+            [class.btn-opacity-disabled]="signUpForm.invalid"
+            [disabled]="(!signUpForm.valid || submitted) && !appError">
             SIGN UP
         </button>
-
+        <div class="form-error-messages" *ngIf="appError">A user with this email address or username already exist!
```

Comment:
Should we specify which between the username and the email  already exists? I feel as though signing up should be as smooth as possible of process.

Reply:
that is the right way, I agree! But for now, we have a generic message for both, in future, we will separate the two after getting this one working properly.